adoc/chapters/programming_interface.adoc

```diff
@@ -1447,7 +1462,11 @@ info::context::atomic_memory_scope_capabilities
 ----
 
     @ [.code]#std::vector<memory_scope>#
-   a@ Returns the set of memory scopes supported by atomic operations on all devices in the context, which is guaranteed to include [code]#work_group#.
+   a@ Returns the set of memory scopes supported by atomic operations on all
+      devices in the context.  When a context returns a "wider" memory scope
+      in this set, it must also return all "narrower" memory scopes.  At a
+      minimum, each context must support [code]#memory_scope::work_item#,
+      [code]#memory_scope::sub_group#, and [code]#memory_scope::work_group#.
```

Comment:
Would it be possible for a device to not have all of them?

Reply:
Again, we are following OpenCL here, which has the same concept of "wide" and "narrow" memory scopes:

> Because atomic scopes are hierarchical, a device that supports a wide scope must also support all narrower scopes, except for the work-item scope, which is a special case.

(https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_ATOMIC_MEMORY_CAPABILITIES)

The new wording does not require support for any "wider" scope than before.  The previous wording also required a device to support `work_group` scope.  The question from our implementation team was whether this meant that the device must also support the narrower scopes.  We thought that we should follow OpenCL in this regard and require support for the narrower scopes too.

Note that we do not adopt the OpenCL language about "except for the work-item scope, which is a special case" because work-item scope has a special meaning in OpenCL related to read/write images.  SYCL does not have that issue because we do not have read/write images, so there is no special semantic regarding `work_item` scope.